lightning/src/ln/channel.rs

```diff
@@ -3679,6 +3679,21 @@ impl<Signer: Sign> Channel<Signer> {
 		}
 	}
 
+	/// Check if the we are trying to reestablish a connection when we have a
+	/// very old local commitment transaction.
+	///
+	/// If we are out of sync we return an empty result, otherwise just the channel error tu sent.
+	fn try_reestablish_when_we_are_late(&mut self,  msg: &msgs::ChannelReestablish) -> Result<(), ChannelError> {
+		//FIXME(vincenzopalazzo) We receive the next commitment transaction to one to sent, so to be in sync we should
+		// be equal at list to the `next - 1`?
+		if (INITIAL_COMMITMENT_NUMBER - self.cur_holder_commitment_transaction_number) < msg.next_remote_commitment_number - 1 {
```

Comment:
Heh, you have an integer underflow on this line.

Reply:
Damnt! Thanks!

Let's see what see the CI now, to check if my actual change fails somewhere (hopefully!)